docs/review.md

```diff
@@ -0,0 +1,579 @@
+```
+
+Also avoid directly setting the DATABASE_URL like so:
+
+```ruby
+ENV['DATABASE_URL'] = 'postgres://localhost/chitter_test'
+DataMapper.setup(:default, ENV['DATABASE_URL'])
+```
+
+# Step 3: Tests and \*\_spec.rb files  
+
+## \*\_spec.rb files (unit tests)
+
+### RSpec features scenarios
+
+Ensure your feature tests look like feature tests, not unit tests.  Unit tests should have only one expect per it block.  Feature test scenarios can have more than one expect; and should have in order to improve comprehensibility and to avoid excessive running times, but don't go crazy.
+
+* good
+
+```ruby
+scenario 'Can create peeps after sign up'  do
+  sign_up_and_create_peep('Test text')
+  expect(current_path).to eq '/peeps'
+  expect(page.status_code).to eq 200
+
+  within 'ul#peeps' do
+    expect(page).to have_content('Test text')
+  end
+end
+```
+
+* not so good
+
+```ruby
+scenario 'Current path is correct after signing up and peeping'  do
+  sign_up_and_create_peep('Test text')
+  expect(current_path).to eq '/peeps'
+end
+
+scenario 'Status code is correct after signing up and peeping'  do
+  sign_up_and_create_peep('Test text')
+  expect(page.status_code).to eq 200
+end
+
+scenario 'Peep is shown after signing up and peeping'  do
+  sign_up_and_create_peep('Test text')
+  within 'ul#peeps' do
+    expect(page).to have_content('Test text')
+  end
+end
+```
+
+
+### Correct location of feature/acceptance and unit tests
+
+Ensure that all your acceptance tests are in a separate folder called `features`.  This can be in your `spec` folder or on the root; up to you.
+
+If a test is in the feature folder it should be testing the entire stack, i.e. it should interact with the app via a web page, and then test the results of that action in the web page that gets returned.  It's acceptable to manipulate the database directly to set things up, but note their are risks associated here, i.e. that you will get your database in a state that it couldn't get into via the web interface, and thus when the tests pass it may not reflect precisely the user experience of using the site.
+
+Conversely if you are testing your models, with or without database interactions, then these tests should NOT be in your feature folder and should be in the `spec` folder, or in `spec/models`
+
+### Need unit tests for datamapper
+
+TODO I think we should encourage the use of a [shoulda](https://github.com/thoughtbot/shoulda-matchers) style syntax for datamapper:
+
+https://github.com/greyblake/dm-rspec
+
+But I realise that this might not be popular - but I think it's an important part of following the acceptance test, unit test cycle ...
+
+Maybe this should be part of moving to something like:
+
+* https://github.com/jeremyevans/sequel
+* https://github.com/kematzy/minitest-sequel
+* https://github.com/openhood/rspec_sequel_matchers
+
+I guess we can't have any of the above when the bookmark_manager project doesn't address any of this ...
```

Comment:
I agree I would love to see our models unit tested (as well as some more unit-tested service objects) but for the moment this is inconsistent with the current curriculum so remove this for the moment